analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -298,16 +405,60 @@ static IEnumerable<TMessage> ReadProtobuf(string path)
                                            OutputKind outputKind,
                                            IEnumerable<MetadataReference> additionalReferences,
                                            string sonarProjectConfigPath = null)
+        {
+            using var scope = new EnvironmentVariableScope(true) { EnableConcurrentAnalysis = true};
+            CreateConcurrencyTest(paths, out var pathsWithConcurrencyTests);
+            var solution = SolutionBuilder.CreateSolutionFromPaths(pathsWithConcurrencyTests, outputKind, additionalReferences);
+            CompileAndVerifyAnalyzer(solution, diagnosticAnalyzers, options, checkMode, sonarProjectConfigPath);
+        }
+
+        private static void CreateConcurrencyTest(IEnumerable<string> paths, out List<string> resultingPaths)
+        {
+            resultingPaths = new List<string>(paths);
+            var language = AnalyzerLanguage.FromPath(paths.First());
+            foreach (var path in paths)
+            {
+                var fullName = new FileInfo(path).FullName;
+                var content = File.ReadAllText(fullName, Encoding.UTF8);
+                if (language == AnalyzerLanguage.CSharp)
+                {
+                    var newPath = $"{fullName}.cs";
```

Comment:
It's not very clear what happend in `Something.cs.cs` by the extra `.cs`. I liked the old name more. Please add something explanatory like `.Concurrent.cs`